comp/otelcol/ddflareextension/def/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/DataDog/datadog-agent/comp/otelcol/ddflareextension/def
 
-go 1.22.0
+go 1.22.7
+
+toolchain go1.23.3
```

Comment:
Ah looks like otel has bumped to go 1.23 !
We should bump the directive in every module too, to avoid the automatic bump which sets a toolchain directive...